RelationalRecord/src/Relations/User.hs

```diff
@@ -0,0 +1,80 @@
+{-# LANGUAGE TemplateHaskell, MultiParamTypeClasses, FlexibleInstances #-}
+{-# LANGUAGE RecordWildCards #-}
+
+module  Relations.User where
+
+import  Types.User as User
+import  Types.DB
+import  Relations.DB
+
+import  Database.Relational.Query
+import  Database.HDBC.Query.TH      (makeRecordPersistableDefault)
+
+import  Data.Int                    (Int32)
+
+
+
+-- SELECTS
+
+allUsers :: Relation () Users
+allUsers = User.users
+
+getUser :: Relation PKey Users
+getUser = relation' . placeholder $ \usrId -> do
+    a       <- query users
+    wheres  $ a ! User.id' .=. usrId
+    return  a
+
+
+-- INSERTS
+
+data UserInsert = UserInsert
+    { iTenantId     :: PKey
+    , iUsername     :: String
+    , iPassword     :: String
+    , iFirstName    :: Maybe String
+    , iLastName     :: Maybe String
+    }
+$(makeRecordPersistableDefault ''UserInsert)
+
+piUser :: Pi Users UserInsert
+piUser = UserInsert
+    |$| User.tenantId'
+    |*| User.username'
+    |*| User.password'
+    |*| User.firstName'
+    |*| User.lastName'
+
+insertUser :: Insert UserInsert
+insertUser = derivedInsert piUser
+
+
+-- UPDATES
+
+data UserUpdate = UserUpdate
```

Comment:
the solution would be to use remove that field from the variadic update type completely.
for those updates who require special attention / a special code-path (you mentioned side effects), a specialized, explicit ```Update``` relation and domain API function would be in order.